tests/test_creator.py

```diff
@@ -197,6 +197,192 @@ def check_parametric_view(spline, conform):
+        # CPTS Manipulation for slightly tangled spline
+        NURBS_eqW_tang.control_points[5] = [3, -3, 1.51]
+        NURBS_eqW_tang.control_points[17] = [3, -3, 1.49]
+
+        # After degree elevation det(J) < 0??
+        NURBS_eqW.elevate_degrees([0, 0, 1, 2])
+
+        # unequal weights:
+        NURBS_uneqW = c.splinepy.NURBS(
+            degrees=[3, 1],
+            knot_vectors=[
+                [0.0, 0.0, 0.0, 0.0, 2.0, 2.0, 4.0, 4.0, 4.0, 4.0],
+                [0.0, 0.0, 1.0, 1.0],
+            ],
+            control_points=[
+                [0.0, 0.0],
+                [0.5, 0.0],
+                [1.0, -1.0],
+                [2.0, -3.0],
+                [2.5, -3.0],
+                [3.0, -3],
+                [5.0, 5.0],
+                [6.0, 5.0],
+                [7.0, 5.0],
+                [8.0, 5.0],
+                [9.0, 5.0],
+                [10.0, 5.0],
+            ],
+            weights=[1, 1, 1, 1.5, 1, 1, 1, 1, 1, 1, 1, 1],
+        )
+        NURBS_uneqW = c.splinepy.helpme.create.extruded(NURBS_uneqW, [0, 0, 3])
+
+        # Splines which are not tangled
+
+        for sp_i in (
+            Bez2,
+            BSpC0,
+            BSpC1,
+            NURBS_eqW,
+            self.bezier_2p2d(),
+            self.bspline_2p2d(),
+            *self.all_3p3d_splines(),
+        ):
+            rng = np.random.default_rng(12345)
+            detSp, d_min = c.splinepy.helpme.create.determinant_spline(
+                sp_i, dmin=True, show=False
+            )
+            rnd_queries = rng.random((10, sp_i.dim), "float32")
+            self.assertTrue(
+                np.allclose(
+                    detSp.evaluate(queries=rnd_queries)[:, 0],
+                    np.linalg.det(sp_i.jacobian(queries=rnd_queries)),
+                )
+            )
+            self.assertTrue(d_min > 0)
+
+        # Splines which are tangled or singular
+        for sp_i in (Bez1, BSpC1_tang, NURBS_eqW_tang):
+            detSp, d_min = c.splinepy.helpme.create.determinant_spline(
+                sp_i, dmin=True, show=False
+            )
+            rnd_queries = rng.random((10, sp_i.dim), "float32")
+            self.assertTrue(
+                np.allclose(
+                    detSp.evaluate(queries=rnd_queries)[:, 0],
+                    np.linalg.det(sp_i.jacobian(queries=rnd_queries)),
+                )
+            )
+            self.assertTrue(d_min <= 0)
+
+        # Expect Failure: rational splines with unequal weights:
+
+        with self.assertRaises(TypeError):
+            detSp = c.splinepy.helpme.create.determinant_spline(NURBS_uneqW)
+
+        with self.assertRaises(TypeError):
+            detSp = c.splinepy.helpme.create.determinant_spline(
+                self.rational_bezier_2p2d()
+            )
+
+        with self.assertRaises(TypeError):
+            detSp = c.splinepy.helpme.create.determinant_spline(
+                self.nurbs_2p2d()
+            )
```

Comment:
If you want to keep these changes, maybe we can add a flag to ignore concerns when approximating nurbs splines ? What does @j042 think?

Reply:
Sure. I think such warnings should be a debug logs and this warning should go into docs

Reply:
Ok so the warnings are now getting logged in the function itself, if a rational spline with unequal weights is given.
Should the test check if this warning is actually logged or is this not neccessary?

Reply:
no, just warning no tests on this matter